sandec/sandec.c

```diff
@@ -168,12 +169,16 @@ Ussage: sandec [filename].raw\n");
    filename[str_len-4] = '.';
    
    fd2 = open(filename, O_CREAT|O_WRONLY|O_BINARY|O_EXCL);
+
    if(fd2<0)
    {
-      printf("Unable to open '%s'!\n", filename);
+      char cwd[512];
+      getcwd(cwd, sizeof(cwd));
```

Comment:
Check the return value of `getcwd`. If it returns NULL then the path exceeded `sizeof(cwd)` with `ERANGE`.

Reply:
Makes sense.